program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+1. Support for bare metal use of C with CVA6 (64-/32-bit) together with the _Newlib_ C library;
+2. Support for Linux application use of C and C++ with CVA6 (32-/64-bit) together with the _GlibC_ C library; and
+3. Support for other RTOSes.
+
+The proposal addresses just the first phase. Other phases will be the subject of separate proposals. Phase 1 will provide the following:
+
+- suitability of Clang/LLVM for HW verification of CVA6:
+  - seamless support of multiple datapath widths (multi-XLEN support);
+  - resolution of functional failures identified on the current CVA6 verification flow.
+
+- support of differentiating characteristics of the CVA6 cores:
+  - pipeline depth,
+  - specific CSRs.
+
+- support for a generic CORE-V instruction set extension interface:
+  - in the Clang/LLVM tool chain.
+  - NOTE: this will provide a commercial driver for future tool chain development for new extensions, whether public or proprietary.
+
+In order to support the Software TG's primary goal of developing a thriving commercial ecosystem, only a basic implementation will be provided.
+By basic implementation we mean that:
+- the assembler/linker will support the instructions;
+- the compiler will have intrinsic (builtin) function support;
+- the compiler will have patterns to generate the instructions from C code in obvious circumstances.
+
+There will be no attempt to provide compiler optimizations.  The development of optimizations will be under the responsibility of:
+- software companies which can rely on the amount of work still needing to be done to drive their businesses; and of
+- HW and system integration companies which can use specific optimizations targeted at their products as market differentiators.
+
+In order to support the Software TG secondary goal of upstreaming all open source tool developments:
+- all Clang/LLVM development will be kept compliant with the LLVM Foundation coding and quality assurance standards;
+- the implementation will follow the upstream tool design and coding conventions; and
+- any vendor-specific modifications or additions will be duly isolated into extensions, so as to maintain a fully functional common open source code base.
+
+Finished LLVM-related work that is not vendor-restricted will be contributed to the LLVM community and maintained upstream.
+
+### Nature of the development
+
+This project requires the modification of a set of existing public open source code bases maintained as projects of the LLVM Foundation and the REMS project.  Therefore, the processes used within OpenHW will reflect the processes of those upstream projects.
+
+The code base of LLVM is of substantial size:
+
+- Clang: 1.75 MLOC C++;
+- LLVM: 2.1 MLOC C++.
+
+The LLVM code base includes substantial regression test suites (1.5 MLOC including `libcxx` C++ library tests), and success with these test suites is a pre-requisite of upstream acceptance of any patch.
+
```

Comment:
LLVM has no execution regression tests, only tests up to IR level. Standard practice is to use GCC regression tests with the LLVM compiler.